test/Libraries/RevitIntegrationTests/WorkflowTests.cs

```diff
@@ -0,0 +1,221 @@
+            var model = ViewModel.Model;
+            Assert.AreEqual(25, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(33, model.CurrentWorkspace.Connectors.Count());
+
+            // Check StructuralFraming.BeamByCurve
+            var beam = "bcef7f1f-fff5-48bc-8e78-7302e7344be8";
+            AssertPreviewCount(beam, 23);
+
+            // get all StructuralFraming.BeamByCurve.
+            for (int i = 0; i < 23; i++)
+            {
+                var allbeam = GetPreviewValueAtIndex(beam, i) as StructuralFraming;
+                Assert.IsNotNull(allbeam);
+            }
+
+            // Check ModelCurve.ByCurve
+            var modelCurve = "8bf46c38-4123-42b6-90ce-35788bf449ce";
+            AssertPreviewCount(modelCurve, 23);
+
+            // get all ModelCurve.ByCurve
+            for (int i = 0; i < 23; i++)
+            {
+                var allstru = GetPreviewValueAtIndex(modelCurve, i) as ModelCurve;
+                Assert.IsNotNull(allstru);
+            }
+        }
+
+        [Test]
+        [TestModel(@".\Workflow\curve.rvt")]
+        public void Test_StructureFrame()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\01_StructuralFraming.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(12, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(12, model.CurrentWorkspace.Connectors.Count());
+
+            //check structuralFraming.BraceByCurve
+            var structuralFraming = "8d0a5991-259c-438d-982a-c9cc2e8b5e79";
+            AssertPreviewCount(structuralFraming, 9);
+
+            //check elements in structuralFraming.BraceByCurve ar not null
+            for (int i = 0; i < 9; i++)
+            {
+                var element = GetPreviewValueAtIndex(structuralFraming, i) as StructuralFraming;
+                Assert.IsNotNull(element);
+            }
+
+        }
+
+
+        [Test]
+        [TestModel(@".\Workflow\DynamoRevit\DynamoSample.rvt")]
+        public void Test_AdaptiveComponent1()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\02_Adaptive Component Placement 1.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(11, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(10, model.CurrentWorkspace.Connectors.Count());
+
+            //check AdaptiveComponent.ByPoint
+            var AC = "357e7a53-361c-4c1e-81ae-83e16213a39a";
+            AssertPreviewCount(AC, 9);
+
+            //check elements in AdaptiveComponent.ByPoint are not null
+            for (int i = 0; i < 9; i++)
+            {
```

Comment:
@line 125 preview count is 9 and in for loop range is from 0 to 9 which is 10 element. How come this test case is passing?